### PR TITLE
SRVKP-6190  new basic step action test

### PR DESCRIPTION
**Pull request by jayesh-garg**
please review @ppitonak Pipeline tested : https://console-openshift-console.apps.cicd.ospqa.com/k8s/ns/jgarg/tekton.dev~v1~PipelineRun/pipeline-with-step-action-6851ky

### CODE REVIEW

File: specs/pipelines/step-action-resolvers.spec [hunk 1]
```diff
@@ -0,0 +1,22 @@
+PIPELINES-32
+# step action resolvers spec
+
+Pre condition:
+  * Validate Operator should be installed
+## Test the functionality of step action resolvers: PIPELINES-32-TC01
+Tags: e2e, sanity
```
Comment (ppitonak) [inline specs/pipelines/step-action-resolvers.spec:7]: add tags `ecosystem` and `non-admin`

**Review by ppitonak — CHANGES_REQUESTED** [review]
1. This spec should be placed to `specs/ecosystem`.
2. This spec is supposed to test step action but it doesn't use it anywhere.
3. We already have a task in `testdata/ecosystem/tasks/git-clone-stepaction` and pipelinerun in `testdata/ecosystem/pipelineruns/git-clone-stepaction` so you can use those, don't create new ones unless there are multiple use-cases that you want to test

File: specs/ecosystem/ecosystem.spec [hunk 2]
```diff
@@ -176,6 +176,24 @@ Steps:
...
+Component: Resolvers
+Level: Integration
+Type: Functional
+Importance: High
+
+Steps:
+    * Verify ServiceAccount "pipeline" exist
+    * Create 
+      |S.NO|resource_dir                                                     |
+      |----|-----------------------------------------------------------------|
```
Comment (ppitonak) [inline specs/ecosystem/ecosystem.spec:190]: You omited creation of task
```
Pipeline releasetest-wjddg/git-clone-stepaction-run can't be Run; it contains Tasks that don't exist: Couldn't retrieve Task "git-clone-stepaction": tasks.tekton.dev "git-clone-stepaction" not found
```

File: specs/ecosystem/ecosystem.spec [hunk 3]
```diff
@@ -176,6 +176,24 @@ Steps:
       |----|-----------------|----------|-----------------------|
       |1   |maven-run        |successful|no                     |
 
+## Test the functionality of step action resolvers: PIPELINES-29-TC10
```
Comment (ppitonak) [inline specs/ecosystem/ecosystem.spec:179]: rebase and update this ID

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: specs/ecosystem/ecosystem.spec (hunk 3 above)
Comment (jayesh-garg) [reply specs/ecosystem/ecosystem.spec]: Done

File: specs/ecosystem/ecosystem.spec (hunk 2 above)
Comment (ppitonak) [reply specs/ecosystem/ecosystem.spec:295]: this is still not addressed

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: specs/ecosystem/ecosystem.spec (hunk 2 above)
Comment (jayesh-garg) [reply specs/ecosystem/ecosystem.spec:295]: Done

File: specs/ecosystem/ecosystem.spec [hunk 4]
```diff
@@ -281,5 +281,21 @@ Steps:
...
+Type: Functional
+Importance: High
+
+Steps:
+    * Verify ServiceAccount "pipeline" exist
+    * Create 
+      |S.NO|resource_dir                                                     |
+      |----|-----------------------------------------------------------------|
+      |1   |testdata/ecosystem/tasks/git-clone-stepaction.yaml               |
+      |2   |testdata/ecosystem/pipelineruns/git-clone-stepaction.yaml        |
```
Comment (ppitonak) [inline specs/ecosystem/ecosystem.spec:297]: Pipeline run is stuck
```
0/5 nodes are available: persistentvolumeclaim "shared-pvc" not found. preemption: 0/5 nodes are available: 5 Preemption is not helpful for scheduling.
```

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: specs/ecosystem/ecosystem.spec (hunk 4 above)
Comment (jayesh-garg) [reply specs/ecosystem/ecosystem.spec:297]: Done
`2024/11/28 10:32:08 Waiting for operator to be up and running....
2024/11/28 10:32:09 Operator is up
      * Validate Operator should be installed	 ...[PASS]
2024/11/28 10:32:09 Verify SA "pipeline" is created in namespace "releasetest-rx4j6"
      * Verify ServiceAccount "pipeline" exist	 ...[PASS]
2024/11/28 10:32:11 output: task.tekton.dev/git-clone-stepaction created
2024/11/28 10:32:12 output: pipelinerun.tekton.dev/git-clone-stepaction-run created
2024/11/28 10:32:13 output: persistentvolumeclaim/shared-pvc created
      * Create 
      
         |S.NO|resource_dir                                             |
         |----|---------------------------------------------------------|
         |1   |testdata/ecosystem/tasks/git-clone-stepaction.yaml       |
         |2   |testdata/ecosystem/pipelineruns/git-clone-stepaction.yaml|
         |3   |testdata/pvc/pvc.yaml                                    |	 ...[PASS]
2024/11/28 10:32:13 validating pipeline run git-clone-stepaction-run for success state...
2024/11/28 10:32:48 pipelineRun: git-clone-stepaction-run is successful under namespace : releasetest-rx4j6
      * Verify pipelinerun 
      
         |S.NO|pipeline_run_name       |status    |check_label_propagation|
         |----|------------------------|----------|-----------------------|
         |1   |git-clone-stepaction-run|successful|no                     |	 ...[PASS]
2024/11/28 10:32:56 output: project.project.openshift.io "releasetest-rx4j6" deleted

Successfully generated xml-report to => /Users/jayesh/project/release-tests/reports/xml-report

Transformed SuiteResult to report structure
Successfully generated html-report to => /Users/jayesh/project/release-tests/reports/html-report/index.html

Done generating HTML report using theme from /Users/jayesh/.gauge/plugins/html-report/4.3.1/themes/default
Sending kill message to Html Report plugin.
Sending kill message to Xml Report plugin.
Sending kill message to runner.
ok  	command-line-arguments	54.703s
Runner with PID:30798 has exited
Specifications:	1 executed	1 passed	0 failed	0 skipped
Scenarios:	1 executed	1 passed	0 failed	0 skipped

Total time taken: 52.039s
Updates are available. Run `gauge update -c` for more info.`

Comment (ppitonak) [reply specs/ecosystem/ecosystem.spec:297]: While this will work, it is not in logical order... create PVC before creating pipelinerun

**Review by ppitonak — CHANGES_REQUESTED** [review]

File: specs/ecosystem/ecosystem.spec (hunk 4 above)
Comment (jayesh-garg) [reply specs/ecosystem/ecosystem.spec]: Done

**Review by ppitonak — APPROVED** [review]

**Pull request merged**